Build/Scripts/EnlistmentHelperFunctions.psm1

```diff
@@ -17,4 +17,86 @@ function Get-BuildMode() {
         return $ENV:BuildMode
     }
     return 'Default'
-}
\ No newline at end of file
+}
+
+<#
+.Synopsis
+    Get the value of a key from the BuildConfig.json or AL-GO-Settings file
+.Parameter ConfigType
+    The type of config file to read from. Can be either "BuildConfig" or "AL-GO"
+.Parameter Key
+    The key to read the value from
+#>
+function Get-ConfigValueFromKey() {
+    param(
+        [Parameter(Mandatory=$false)]
+        [ValidateSet("BuildConfig","AL-GO")]
+        [string]$ConfigType = "AL-GO",
+        [Parameter(Mandatory=$true)]
+        [string]$Key
+    )
+
+    if ($ConfigType -eq "BuildConfig") {
+        $ConfigPath = Join-Path (Get-BaseFolder) "Build/BuildConfig.json" -Resolve
+    } else {
+        $ConfigPath = Join-Path (Get-BaseFolder) ".github/AL-Go-Settings.json" -Resolve
+    }
+    $BuildConfig = Get-Content -Path $ConfigPath -Raw | ConvertFrom-Json
+    return $BuildConfig.$Key
+}
+
+<#
+.Synopsis
+    Get the nuget.exe if it doesn't exist
+#>
+function Get-NugetExe() {
```

Comment:
Just suggestions:

`Download-NugetExe`
`Restore-NugetExe`

Reply:
I like Download. Think I got carried away with Get and Set :D